pkg/util/system/cgroup.go

```diff
@@ -46,26 +45,25 @@ type CPUStatRaw struct {
 	ThrottledNanoSeconds int64
 }
 
-func CgroupFileWriteIfDifferent(cgroupTaskDir string, file CgroupFile, value string) error {
+func CgroupFileWriteIfDifferent(cgroupTaskDir string, file Resource, value string) error {
 	currentValue, currentErr := CgroupFileRead(cgroupTaskDir, file)
 	if currentErr != nil {
 		return currentErr
 	}
 	if value == currentValue || value == CgroupMaxValueStr && currentValue == CgroupMaxSymbolStr {
 		// compatible with cgroup valued "max"
-		klog.V(6).Infof("read before write %s %s and got str value, considered as MaxInt64", cgroupTaskDir,
-			file.ResourceFileName)
+		klog.V(6).Infof("read before write %s and got str value, considered as MaxInt64", file.Path(cgroupTaskDir))
 		return nil
 	}
 	return CgroupFileWrite(cgroupTaskDir, file, value)
 }
 
-func CgroupFileReadInt(cgroupTaskDir string, file CgroupFile) (*int64, error) {
-	if file.IsAnolisOS && !HostSystemInfo.IsAnolisOS {
-		return nil, fmt.Errorf("read cgroup config : %s fail, need anolis kernel", file.ResourceFileName)
+func CgroupFileReadInt(cgroupTaskDir string, r Resource) (*int64, error) {
+	if supported, msg := r.IsSupported(); !supported {
+		return nil, fmt.Errorf("read cgroup %s failed, msg: %s", r.ResourceType(), msg)
 	}
 
-	dataStr, err := CgroupFileRead(cgroupTaskDir, file)
+	dataStr, err := CgroupFileRead(cgroupTaskDir, r)
 	if err != nil {
```

Comment:
// TODO ignore not exist error(not supported)
so do the followings

Reply:
the existence should be checked in `IsSupported()`. If a supported cgroup file is missing, we may report the error.